gitgud/levels/util.py

```diff
@@ -194,12 +192,40 @@ def test_level(level, test):
 
     return True
 
-
-class Level:
+class NamedList:
+    # Pass in list of tuples in format (reference, Object), where reference is of type str
+    def __init__(self, data=[]):
+        self.namedict = {d[0]:i for i, d in enumerate(data)}
+        self.items = [obj for name, obj in data]
+    
+    def __getitem__(self, query):
+        if isinstance(query, int):
+            return self.items[query]
+        elif isinstance(query, str):
+            return self.items[self.namedict[query]]
+        else:
+            return None
+    def __iter__(self):
+        return self.items.__iter__()
+    def __len__(self):
+        return len(self.items)
+    def __setitem__(self, key, item):
+        if isinstance(key, str):
+            self.namedict[key] = len(self.items)
+            self.items.append(item)
+        else:
+            raise TypeError
+    
+    def values(self):
+        return self.items
+    def keys(self):
+        return self.namedict.keys()
+
+class Level():
```

Comment:
Make a new class AllLevels and make NamedList a superclass of both Level and AllLevels